examples/fit_with_bounds.py

```diff
@@ -61,58 +73,85 @@ def gen_from_expr(self,data, guesses={}):
                 "axis names are",
                 axis_names,
                 "variable names are",
-                variable_names,
+                self.variable_names,
                 "parameter names are",
                 parameter_names,
             )
         )
         # }}}
-        pars = Parameters()
-        for this_name in parameter_names:
-            kwargs = {}
-            if this_name in guesses.keys():
-                logger.debug(strm("applying bounds for", this_name))
-                kwargs.update(guesses[str(this_name)])
-            pars.add(this_name, **kwargs)
-        for j in pars:
-            logger.info(strm("fit param ---", j))
-        logger.info(strm(pars))
         self.fn = lambdify(
             variable_symbols + parameter_symbols,
             self.expression,
             modules=[{"ImmutableMatrix": np.ndarray}, "numpy", "scipy"],
         )
-        return pars
+        self.pars = Parameters()
+        for this_name in parameter_names:
+            self.pars.add(this_name)
+    def set_guess(self,*args,**kwargs):
```

Comment:
in master we did this inside the main class method,  rather than making an entirely new method of the set_guesses. Do we want a new method for this? which do you think is the better option?

Reply:
How would you otherwise plan on setting the guess of the instance (this is not a rhetorical question — maybe you’re thinking of something I have not thought of)

Reply:
guesses was a dictionary kwarg that one of the methods took in
```
def gen_from_expr(self,data, guesses={}):
    ...
    for this_name in parameter_names:
            kwargs = {}
            if this_name in guesses.keys():
                logger.debug(strm("applying bounds for", this_name))
                kwargs.update(guesses[str(this_name)])
            pars.add(this_name, **kwargs)

```